docs/understand/texture_fetching.rst

```diff
@@ -0,0 +1,151 @@
+
+This following image shows a texture stretched out to a 4x4 pixel quad, but still indexed in the [0 to 1] range. The in-between values are interpolated between the neighboring texels.
+
+.. figure:: ../data/understand/textures/linear.png
+  :width: 300
+  :alt: Texture upscaled with linear filtering
+  :align: center
+
+  Texture upscaled with linear filtering
+
+Texture addressing
+===============================================================================
+
+Texture addressing mode handles the index that is out of bounds of the texture. This mode describes which values of the texture or a preset value to use when the index is out of bounds.
+
+The following sections describe the various texture addressing methods.
+
+.. _texture_fetching_border:
+Address mode border
+-------------------------------------------------------------------------------
+
+In this method, the texture fetching returns a border value when indexing out of bounds. This must be set before texture fetching.
+
+The following image shows the texture on a 4x4 pixel quad, indexed in the [0 to 3] range. The out-of-bounds values are the border color, which is yellow.
+
+.. figure:: ../data/understand/textures/border.png
+  :width: 300
+  :alt: Texture with yellow border color
+  :align: center
+
+  The purple lines are not part of the texture, only denote the edge, where the addressing begins.
+
+.. _texture_fetching_wrap:
+Address mode wrap
+-------------------------------------------------------------------------------
+
+In this addressing mode, the modulo of the index is calculated:
+
+``tex(x) = T[x mod (size-1)]``
+
+This creates a repeating image effect.
+
+The following image shows the texture on a 4x4 pixel quad, indexed in the [0 to 3] range. The out-of-bounds values are repeating the original texture.
+
+.. figure:: ../data/understand/textures/wrap.png
+  :width: 300
+  :alt: Texture with wrap addressing
+  :align: center
+
+  The purple lines are not part of the texture, only denote the edge, where the addressing begins.
+
+.. _texture_fetching_mirror:
+Address mode mirror
+-------------------------------------------------------------------------------
+
+Similar to the wrap mode, this mode also creates a repeating image, but by mirroring the neighboring instances.
+
+The following image shows the texture on a 4x4 pixel quad, indexed in the [0 to 3] range. The out-of-bounds values are repeating the original texture, but mirrored.
+
+.. figure:: ../data/understand/textures/mirror.png
+  :width: 300
+  :alt: Texture with mirror addressing
+  :align: center
+
+  The purple lines are not part of the texture, only denote the edge, where the addressing begins.
+
```

Comment:
The purple lines are not part of the texture. They only denote the edge, where the addressing begins.